File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        cnt = 0
        result = [[1]]
        while len(result) != numRows:
            br = []
            br.append(result[cnt][0])
            for i in range(len(result[cnt])-1):
                br.append(result[cnt][i] + result[cnt][i+1])
            br.append(result[cnt][-1])
            result.append(br)
            cnt +=1
        return result

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();

        // 첫 번째 줄은 무조건 1
        if (numRows >= 1) {
            List<Integer> firstRow = new ArrayList<>();
            firstRow.add(1);
            triangle.add(firstRow);
        }

        for (int i = 1; i < numRows; i++) {
            List<Integer> prevRow = triangle.get(i - 1);
            List<Integer> row = new ArrayList<>();

            // 각 행의 첫 번째 숫자는 1
            row.add(1);

            // 중간 숫자들 계산
            for (int j = 1; j < i; j++) {
                row.add(prevRow.get(j - 1) + prevRow.get(j));
            }

            // 각 행의 마지막 숫자는 1
            row.add(1);

            triangle.add(row);
        }

        return triangle;
    }
}